### PR TITLE
Add google cloud function rule

### PR DESCRIPTION
**Pull request by Morganamilo**

### CODE REVIEW

**Comment by Morganamilo** [comment]
The name of this rule is `google function` to be a bit more brief. `http google cloud function` seemed like a mouth full. Any preference.

**Comment by Morganamilo** [comment]
This breaks if you have a region with a - in it due to a bug in the control api. Nothing we can do about it here.

**Comment by graham-russell** [comment]
> The name of this rule is `google function` to be a bit more brief. `http google cloud function` seemed like a mouth full. Any preference.

My preference is the same as yours. I've dropped the `http` bit on rules that I've worked on.

File: internal/provider/resource_ably_rule_http_google_cloud_function.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				Description: "JSON provides a text-based encoding",
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					DefaultAttribute(types.String{Value: "json"}),
+				},
+			},
+		},
+
```
Comment (graham-russell) [inline internal/provider/resource_ably_rule_http_google_cloud_function.go:62]: Is the blank line expected?

**Review by graham-russell — APPROVED** [review]

File: internal/provider/resource_ably_rule_http_google_cloud_function.go (hunk 1 above)
Comment (Morganamilo) [reply internal/provider/resource_ably_rule_http_google_cloud_function.go]: Nope :P

**Pull request merged**